src/app-layout/visual-refresh-toolbar/index.tsx

```diff
@@ -432,16 +435,41 @@ const AppLayoutVisualRefreshToolbar = React.forwardRef<AppLayoutProps.Ref, AppLa
       placement.inlineSize,
     ]);
 
+    /**
+     * Returns true if the AppLayout is nested
+     * Does not apply to iframe
+     */
+    const getIsNestedInAppLayout = (element: HTMLElement | null): boolean => {
+      let currentElement: Element | null = element?.parentElement ?? null;
+
+      while (currentElement) {
+        if (getComputedStyle(currentElement).getPropertyValue(globalVars.stickyVerticalTopOffset)) {
+          return true;
+        }
+        currentElement = currentElement.parentElement;
+      }
+
+      return false;
+    };
+
+    useLayoutEffect(() => {
+      setIsNested(getIsNestedInAppLayout(rootRef.current));
+    }, []);
```

Comment:
@just-boris you previously suggested to skip this calculation if toolbar is empty, but it turns out this performance optimization might even backfire if we put it as a dependency here, since it will be recalculated every time this prop changes, so I decided to leave it as is

Reply:
I do not understand this comment. Could you post here in the comments a possible code how you planned to handle the toolbar and which dependency concerned you?

Reply:
```javascript
useLayoutEffect(() => {
      if (toolbarProps) {
            setIsNested(getIsNestedInAppLayout(rootRef.current));
      }
    }, [toolbarProps]);
```

Reply:
How about using `hasToolbar` value. It is boolean and rarely changes: https://github.com/cloudscape-design/components/blob/d1ff27eaa1896e77cdc87fd7b4c3c67528962133/src/app-layout/visual-refresh-toolbar/index.tsx#L291